src/test/java/net/openhft/chronicle/bytes/HeapByteStoreTest.java

```diff
@@ -48,4 +53,35 @@ public void testElasticBytesEnsuringCapacity() {
         bytes.prewriteInt(1);
         assertTrue(bytes.realCapacity()> initialCapacity);
     }
+
+    @Test
+    public void testWriteOnHeap() throws Exception {
+        doTestWrite(() -> ByteBuffer.allocate(128));
+    }
+
+    @Test
+    public void testWriteDirect() throws Exception {
+        doTestWrite(() -> ByteBuffer.allocateDirect(128));
+    }
+
+    private void doTestWrite(Callable<ByteBuffer> generator) throws Exception {
```

Comment:
consider moving this test to BytesTest so it this behaviour is checked for all Bytes types